### PR TITLE
Ex6

### PR DESCRIPTION
**Pull request by zyahav**

### CODE REVIEW

File: ex1/ex1.js [hunk 1]
```diff
@@ -6,3 +6,29 @@
  *     fn(",,,") === 2
  *     fn("hel,lowo,rld") === -1
  */
+
```
Comment (myok12) [inline ex1/ex1.js]: Please revert all files except ex6. Next time, please create the branch from master (which has none of the new files).
I can explain about this over the phone.

File: ex6/ex6.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+				}
+
+			}
+			// We have to push the last tmpStr
+			strArray.push(tmpStr);
+		}
+	}
+	//  Return the array
+	//return strArray;
```
Comment (myok12) [inline ex6/ex6.js]: You forgot the return, all tests probably failing.

File: ex6/ex6.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * split('hello,,,world') === ['hello', '', '', 'world']
+ * split('') === []
+ *
+ */
+function split(str) {
+
+	var strArray = [], tmpStr = '';
+
+	// If no chars in string no work to do so will return strArray empty array
+	if(str != '') {
```
Comment (myok12) [inline ex6/ex6.js]: I like to handle these cases explicitly, i.e.:

> if (str === '') {
>   return [];
>}

This leaves the rest of the code clean and without nested if statements.

File: ex6/ex6.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+		if (str.indexOf(',', 0) === -1) {
```
Comment (myok12) [inline ex6/ex6.js]: It looks like this is redundant (not needed) after looking at the code below.

File: ex6/ex6Spec.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * split('hello') === ['hello']
+ * split('hello,,,world') === ['hello', '', '', 'world']
+ * split('') === []
+ */
+var ex6 = require('./ex6.js');
+
+describe("split", function() {
+
+	it("returns 3 chracters", function() {
+		expect(ex6.split('a,b,c')).toBe('a,b,c');
```
Comment (myok12) [inline ex6/ex6Spec.js]: The function should return an array, not a string

File: ex6/ex6Spec.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * split('hello,world') === ['hello', 'world']
+ * split('hello') === ['hello']
+ * split('hello,,,world') === ['hello', '', '', 'world']
+ * split('') === []
+ */
+var ex6 = require('./ex6.js');
+
+describe("split", function() {
+
+	it("returns 3 chracters", function() {
```
Comment (myok12) [inline ex6/ex6Spec.js]: 3 characters --> 3 elements

File: ex6/ex6Spec.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+	it("returns 2 words", function() {
+		expect(ex6.split('hello,world')).toBe('hello,world');
+	});
+
+	it("returns one word", function() {
+		expect(ex6.split('hello')).toBe('hello');
+	});
+
+	it("return sam words and empty index's", function() {
```
Comment (myok12) [inline ex6/ex6Spec.js]: works with empty strings

File: ex6/ex6Spec.js (hunk 5 above)
Comment (myok12) [inline ex6/ex6Spec.js]: Fix this for all tests.

File: ex6/ex6.js (hunk 4 above)
Comment (zyahav) [inline ex6/ex6.js]: done

File: ex6/ex6Spec.js (hunk 7 above)
Comment (zyahav) [inline ex6/ex6Spec.js]: done

File: ex6/ex6Spec.js (hunk 5 above)
Comment (zyahav) [inline ex6/ex6Spec.js]: done

File: ex6/ex6.js (hunk 2 above)
Comment (myok12) [inline ex6/ex6.js]: You can use `.toInclude()` or `.toNotInclude()` for individual items, or `.toEqual([1,2,3])`. 

https://jasmine.github.io/1.3/introduction#section-Included_Matchers

**Pull request closed without merging**